File: frontend/lib/api/pathfinder.ts
import { api } from "./client";

export interface RecommendedProject {
  title: string;
  description: string;
}

export interface CareerGoal {
  id: string;
  title: string;
  description: string;
  required_skills: string[];
  recommended_projects: RecommendedProject[];
  milestones: string[];
}

export interface SkillGap {
  skill: string;
  status: "ready" | "weak" | "missing";
}

export interface CareerPlan {
  career_goal: CareerGoal;
  skill_gaps: SkillGap[];
  ready_skills: string[];
  missing_skills: string[];
  readiness_score: number;
  recommended_projects: RecommendedProject[];
  learning_path: string[];
  summary: string;
}

export async function fetchCareerGoals(): Promise<CareerGoal[]> {
  return api<CareerGoal[]>("/api/pathfinder/goals");
}

export async function analyzeCareerPath(careerGoalId: string): Promise<CareerPlan> {
  return api<CareerPlan>("/api/pathfinder/analyze", {
    method: "POST",
    body: JSON.stringify({ career_goal_id: careerGoalId }),
  });
}

File: backend/app/models/pathfinder.py
from __future__ import annotations

from pydantic import BaseModel, Field


class RecommendedProject(BaseModel):
    title: str
    description: str


class CareerGoal(BaseModel):
    id: str
    title: str
    description: str
    required_skills: list[str] = Field(default_factory=list)
    recommended_projects: list[RecommendedProject] = Field(default_factory=list)
    milestones: list[str] = Field(default_factory=list)


class SkillGap(BaseModel):
    skill: str
    status: str = "missing"  # "ready" | "weak" | "missing"


class CareerPlan(BaseModel):
    career_goal: CareerGoal
    skill_gaps: list[SkillGap] = Field(default_factory=list)
    ready_skills: list[str] = Field(default_factory=list)
    missing_skills: list[str] = Field(default_factory=list)
    readiness_score: float = 0.0  # 0.0 to 1.0
    recommended_projects: list[RecommendedProject] = Field(default_factory=list)
    learning_path: list[str] = Field(default_factory=list, description="Ordered list of skills to learn")
    summary: str = ""
